client/alloc_runner.go

```diff
@@ -476,33 +473,62 @@ func (r *AllocRunner) Update(update *structs.Allocation) {
 	}
 }
 
-// StatsReporter returns an interface to query resource usage statistics of an
-// allocation
-func (r *AllocRunner) StatsReporter() AllocStatsReporter {
-	return r
-}
-
-// AllocStats returns the stats reporter of all the tasks running in the
-// allocation
-func (r *AllocRunner) AllocStats() map[string]TaskStatsReporter {
+// LatestAllocStats returns the latest allocation stats. If the optional taskFilter is set
+// the allocation stats will only include the given task.
+func (r *AllocRunner) LatestAllocStats(taskFilter string) (*cstructs.AllocResourceUsage, error) {
```

Comment:
@dadgar I think we should keep the interface, so that we can add methods around histograms etc, otherwise we will have to add those methods in the client as well.